test/models/deploys_test.rb

```diff
@@ -538,6 +538,57 @@ def generate_commits(amount:, stack_id:, user_id:, validate:)
       assert_equal commit_ids[1], test_stack.last_deployed_commit.id
     end
 
+    test "#trigger_revert skips deploys from other stacks" do
+      # The revert functionality should only consider Shipit::Deploy and Shipit::Rollback when selecting a target to roll back to for the user.
+      # But it is possible for other task types to be defined, so we want to ensure that they are properly skipped, as we can't know whether they are 'valid' to roll back to.
+      user_id = @user.id
+      test_stack = create_test_stack
+      test_stack.save
+      test_stack.reload
```

Comment:
Is this reload necessary?

Reply:
Ah, I don't seem to re-use test_stack below this point, so it looks like it can be safely removed.